src/frontend/src/handler/create_source.rs

```diff
@@ -68,6 +69,36 @@ async fn extract_avro_table_schema(
         .collect_vec())
 }
 
+/// Map an Avro schema to a relational schema. And extract primary key columns.
+async fn extract_upsert_avro_table_schema(
+    schema: &AvroSchema,
+    with_properties: &HashMap<String, String>,
+) -> Result<(Vec<ColumnCatalog>, Vec<ColumnId>)> {
+    let conf = AvroParserConfig::new(
+        with_properties,
+        schema.row_schema_location.0.as_str(),
+        schema.use_schema_registry,
+        true,
+    )
+    .await?;
+    let vec_column_desc = conf.map_to_columns()?;
+    let mut vec_pk_desc = conf.extract_pks()?;
+    let pks = vec_pk_desc
+        .drain(..)
+        .filter_map(|desc| vec_column_desc.iter().position(|x| x.name == desc.name))
```

Comment:
column_id may not equals to the position, you should use the `column_id` field of `ColumnDesc`, and if we can't find one pk column in `vec_column_desc`, we should return an error, instead of filtering it out directly.

Reply:
done